tests/test_change_compute.py

```diff
@@ -0,0 +1,318 @@
+# CrateDB Kubernetes Operator
+# Copyright (C) 2021 Crate.IO GmbH
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import logging
+from typing import Set
+from unittest import mock
+
+import pytest
+from kubernetes_asyncio.client import CoreV1Api, CustomObjectsApi
+
+from crate.operator.change_compute import generate_body_patch
+from crate.operator.constants import API_GROUP, RESOURCE_CRATEDB
+from crate.operator.cratedb import connection_factory
+from crate.operator.webhooks import WebhookChangeComputePayload
+
+from .utils import (
+    DEFAULT_TIMEOUT,
+    assert_wait_for,
+    cluster_routing_allocation_enable_equals,
+    create_test_sys_jobs_table,
+    is_cluster_healthy,
+    is_kopf_handler_finished,
+    start_cluster,
+)
+
+
+async def do_pods_exist(core: CoreV1Api, namespace: str, expected: Set[str]) -> bool:
+    pods = await core.list_namespaced_pod(namespace=namespace)
+    return expected.issubset({p.metadata.name for p in pods.items})
+
+
+async def do_pod_ids_exist(core: CoreV1Api, namespace: str, pod_ids: Set[str]) -> bool:
+    pods = await core.list_namespaced_pod(namespace=namespace)
+    return bool(pod_ids.intersection({p.metadata.uid for p in pods.items}))
+
+
+@pytest.mark.k8s
```

Comment:
While it's great that we test multiple possible cases, these tests take circa 2 minutes each and add _significantly_ to the overall test duration... Perhaps worth testing just one of the scenarios? WDYT?